### PR TITLE
Allow user defines the bar chart bins

### PR DESCRIPTION
**Pull request by zhx828**
### What & Why
This fixes https://github.com/cbioportal/cbioportal/issues/5819

### Update bins
Aug-14-2019 10-22-04

### When bin definition is invalid
Aug-14-2019 10-22-59

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -1299,7 +1299,27 @@ export class StudyViewPageStore {
             !this._clinicalDataBinFilterSet.get(uniqueKey)!.disableLogScale;
     }
 
-    @action addCharts(visibleChartIds:string[]) {
+    @autobind
+    @action
+    public updateCustomBins(uniqueKey: string, bins: number[]) {
+        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Here is the observable map we discussed.

File: src/pages/studyView/chartHeader/ChartHeader.tsx [hunk 2]
```diff
@@ -139,20 +143,45 @@ export class ChartHeader extends React.Component<IChartHeaderProps, {}> {
             items.push(
                 <li>
                     <a className="dropdown-item"
+                       style={{display: 'flex'}}
```
Comment (alisman) [inline src/pages/studyView/chartHeader/ChartHeader.tsx]: is it necessary?

**Comment by alisman** [comment]
@zhx828 think maybe a single line of instructional text would help above the text area.   also, @jjgao, i experienced some funky behavior with the flyout from the flyout behavior.  hard to reproduce.  the bins flyout switched alignment.   a simpler, most robust UI would be to open it in a modal dialog.   the drawback would be that the user couldn't see the underlying chart for reference.  but that's true even now when the dialog opens to the left.   
image

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (alisman) [reply src/pages/studyView/StudyViewPageStore.ts]: @zhx828 we should examine why this causes entire studyview to reload.  i think this the the only action which takes place, right?  i agree with @jjgao that this should really not result in a reload.   i would be interested in getting to the bottom of it with you

**Comment by jjgao** [comment]
I think a modal dialog may be a good idea in this case. Some description would also be good.

**Comment by zhx828** [comment]
@jjgao @alisman I updated the PR with modal implementation. The gif above has been updated. You can also check the feather here https://cbioportal-frontend-pr-2518.herokuapp.com/study/summary?id=acc_tcga_pan_can_atlas_2018
Let me know what you think.

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (zhx828) [reply src/pages/studyView/StudyViewPageStore.ts]: This issue has been resolved.

**Comment by zhx828** [comment]
The failing e2e is not related to this pr.

**Pull request merged**